src/profile/notification-settings/PreferenceCheckbox.tsx

```diff
@@ -0,0 +1,65 @@
+import _ from 'lodash/fp';
+import React from 'react';
+import { Checkbox } from 'src/components/common';
+import { withErrorReporting } from 'src/libs/error';
+import { EventWorkspaceAttributes } from 'src/libs/events';
+import * as Utils from 'src/libs/utils';
+import {
+  notificationEnabled,
+  NotificationType,
+  updateNotificationPreferences,
+  updateUserAttributes,
+} from 'src/profile/notification-settings/utils';
+
+export interface NotificationCheckboxProps {
+  notificationKeys: string[];
+  notificationType: NotificationType;
+  label: string;
+  setSaving: (boolean) => void;
+  prefsData: Record<string, string>;
+  workspace: EventWorkspaceAttributes | undefined;
+}
+
+export const NotificationCheckbox = (props: NotificationCheckboxProps) => {
```

Comment:
Taken from https://github.com/DataBiosphere/terra-ui/blob/366a30ae3ae2b263355581e63a45de9eebc8017c/src/profile/notification-settings/NotificationSettings.js#L17, converted to TSX and Ajax and "enabled" logic moved to a utils file so that it can be shared with the WorkspaceDashboard code.